utils/patch-release.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Script: patch-release.sh
+# Description:
+#   Cherry-picks a list of commits, amends each with the original commit hash for tracking,
+#   and generates a summary from the short github commit messages with links to each commit.
+#
+# Usage:
+#   ./patch-release.sh <commit1> <commit2> ...
+#
+# Example:
+#   ./patch-release.sh abc1234 def5678
+
+set -e
+
+# Check if at least two arguments are provided (repo and at least one commit)
+if [ "$#" -lt 1 ]; then
+    echo "Usage: $0 <commit1> <commit2> ..."
+    echo "Example: $0 abc1234 def5678"
+    exit 1
+fi
+
+GITHUB_REPO="aiidateam/aiida-core"
+
+# Create an array to store commit summaries
+declare -a commit_summaries=()
+
+# Loop through each commit hash
+for commit in "$@"; do
+    # Cherry-pick the commit
+    if git cherry-pick "$commit"; then
+        # If cherry-pick succeeds, get the short message and short hash
+        commit_message=$(git log -1 --pretty=format:"%B" HEAD)
+        original_short_hash=$(git log -1 --pretty=format:"%h" "$commit")
+        original_long_hash=$(git rev-parse $original_short_hash)
+
+        # Amend the cherry-picked commit to include the original commit ID for tracking
+        git commit --amend -m "$commit_message" -m "Cherry-pick: $original_long_hash"
+
+        # Format the output as a Markdown list item and add to the array
+        short_commit_message=$(git log -1 --pretty=format:"%s" HEAD)
+        cherry_picked_hash=$(git log -1 --pretty=format:"%h" HEAD)
+        commit_summaries+=("- $short_commit_message [[${commit}]](https://github.com/$GITHUB_REPO/commit/${original_long_hash})")
+    else
+        echo "Failed to cherry-pick commit $commit"
```

Comment:
Maybe this message could be extended to say something about possible next steps? But I am not sure.

Reply:
I am also not sure what should be the next step^^ Read the error message and try to fix it is a bit too generic to be useful.

Reply:
Agree. Also I guess the person using this script should know what their doing. :sweat_smile: